docs/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/migrate-to-an-out-of-tree-cloud-provider/migrate-to-out-of-tree-azure.md

```diff
@@ -0,0 +1,205 @@
+---
+title: Migrating Azure In-tree to Out-of-tree
+---
+
+<head>
+  <link rel="canonical" href="https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/migrate-to-an-out-of-tree-cloud-provider/migrate-to-out-of-tree-azure"/>
+</head>
+
+Kubernetes is moving away from maintaining cloud providers in-tree. In Kubernetes v1.30 and later, the in-tree cloud providers have been removed. Rancher v2.9.0 allows you to upgrade to Kubernetes v1.30 when you migrate from an in-tree to out-of-tree provider.
```

Comment:
> In Kubernetes v1.30 and later, the in-tree cloud providers have been removed.

In the AWS doc, we have:

> In Kubernetes v1.27 and later, the in-tree cloud providers have been removed.

I checked the Kubernetes official repo and it looks like they keep moving the milestone out...maybe we shouldn't mention a version number at all?

See:
- https://github.com/kubernetes/enhancements/issues/2395#issuecomment-841085779
- https://github.com/kubernetes/enhancements/issues/2395#issuecomment-1894679080
- https://github.com/kubernetes/enhancements/issues/2395#issuecomment-2061706417

Reply:
Since they keep moving their milestone, we also need to update the vSphere migration docs, since we assumed a 1.27 release. I can do that in another PR

Reply:
@martyav Versions changing depending on cloud provider, for instance it got removed for azure >=1.30 but was removed for amazon >=1.27. Since this is azure file, I mentioned 1.30. I'll update the lines to clarify azure specifically to make it clear.

Reply:
For reference, the [1.30 changelog](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.30.md) calls out:
- `The in-tree cloud provider for Azure has now been removed. Please use the external cloud provider and CSI driver from https://github.com/kubernetes/cloud-provider-azure instead. (https://github.com/kubernetes/kubernetes/pull/122857, [@nilo19](https://github.com/nilo19))`
- `The in-tree cloud provider for vSphere has been deprecated and removed. Users are advised to utilize the external cloud provider and CSI driver available at https://github.com/kubernetes/cloud-provider-vsphere. (https://github.com/kubernetes/kubernetes/pull/122937, [@dims](https://github.com/dims))`
 

The [1.27 changelog](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.27.md) calls out:
- `The in-tree cloud provider for AWS (and the EBS storage plugin) has now been removed. Please use the external cloud provider and CSI driver from https://github.com/kubernetes/cloud-provider-aws instead. (https://github.com/kubernetes/kubernetes/pull/115838, [@torredil](https://github.com/torredil)) [SIG API Machinery, Apps, Architecture, Auth, CLI , Cloud Provider, Cluster Lifecycle, Instrumentation, Node, Release, Scheduling, Storage, and Testing]`

Reply:
Thanks for confirming, I'll update the files with the upstream links!